flink-connector-jdbc/pom.xml

```diff
@@ -38,10 +38,12 @@ under the License.
 		<scala.binary.version>2.12</scala.binary.version>
 		<scala-library.version>2.12.7</scala-library.version>
 		<assertj.version>3.23.1</assertj.version>
+		<jackson.version>2.15.2</jackson.version>
 		<postgres.version>42.5.1</postgres.version>
 		<oracle.version>21.8.0.0</oracle.version>
 		<trino.version>418</trino.version>
 		<byte-buddy.version>1.12.10</byte-buddy.version>
+		<elasticsearch.version>8.11.1</elasticsearch.version>
```

Comment:
We need to be really careful about what's included here, since the Elasticsearch dependency that's being included here isn't a 100% Apache license.

Reply:
To be honest I'm not fully aware of all consequences of mixing licences here.

Below I summarize all elastic dependencies which are used:
* `org.elasticsearch.plugin:x-pack-sql-jdbc` - Elastic License 2.0 - The dependency has `provided` scope so it will not be part of the connector "fat-jar". If someone wants to use it, has to add it to classpath/container/etc on their own. Does it make us "safe"?

* `org.elasticsearch.client:elasticsearch-rest-client` (https://mvnrepository.com/artifact/org.elasticsearch.client/elasticsearch-rest-client) used with `test` scope - Apache 2 License - we are safe here.

Reply:
> What should be done to ensure if adding this dependency (even as "provided") is allowed?

I think we should add an enforcer rule to make sure that it's never bundled. Next to that, I think we should make a similar statement in the docs as we've done for RabbitMQ (see https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/datastream/rabbitmq/)

Reply:
@MartijnVisser Both enforcer rule and statement about JDBC driver license have been added.